packages/path_provider/path_provider_macos/example/test_driver/path_provider_e2e.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'dart:io';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:e2e/e2e.dart';
+
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized();
+
+  testWidgets('getTemporaryDirectory', (WidgetTester tester) async {
+    final Directory result = await getTemporaryDirectory();
+    _verifySampleFile(result, 'temporaryDirectory');
+  });
+
+  testWidgets('getApplicationDocumentsDirectory', (WidgetTester tester) async {
+    final Directory result = await getApplicationDocumentsDirectory();
+    _verifySampleFile(result, 'applicationDocuments');
+  });
+
+  testWidgets('getApplicationSupportDirectory', (WidgetTester tester) async {
+    final Directory result = await getApplicationSupportDirectory();
+    _verifySampleFile(result, 'applicationSupport');
+  });
+
+  testWidgets('getLibraryDirectory', (WidgetTester tester) async {
+    if (Platform.isIOS) {
+      final Directory result = await getLibraryDirectory();
+      _verifySampleFile(result, 'library');
+    } else if (Platform.isAndroid) {
+      final Future<Directory> result = getLibraryDirectory();
+      expect(result, throwsA(isInstanceOf<UnsupportedError>()));
+    }
+  });
+
+  testWidgets('getExternalStorageDirectory', (WidgetTester tester) async {
+    if (Platform.isIOS) {
+      final Future<Directory> result = getExternalStorageDirectory();
+      expect(result, throwsA(isInstanceOf<UnsupportedError>()));
+    } else if (Platform.isAndroid) {
+      final Directory result = await getExternalStorageDirectory();
+      _verifySampleFile(result, 'externalStorage');
+    }
```

Comment:
Is this run against all platforms? Shouldn't this test what happens on the macos platform?

Reply:
nope, good catch :)